Detectors/TOF/reconstruction/include/TOFReconstruction/DecoderBase.h

```diff
@@ -0,0 +1,93 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   Decoder.h
+/// @author Roberto Preghenella
+/// @since  2020-02-24
+/// @brief  TOF compressed data decoder
+
+#ifndef O2_TOF_DECODERBASE
+#define O2_TOF_DECODERBASE
+
+#include <fstream>
+#include <string>
+#include <cstdint>
+#include <vector>
+#include "Headers/RAWDataHeader.h"
+#include "DataFormatsTOF/CompressedDataFormat.h"
+
+namespace o2
+{
+namespace tof
+{
+namespace compressed
+{
+
+class DecoderBase
+{
+
+ public:
+  DecoderBase() = default;
+  ~DecoderBase() = default;
+
+  inline bool run()
+  {
+    rewind();
+    while (!processHBF())
+      ;
+    return false;
+  };
+
+  inline void rewind()
+  {
+    decoderRewind();
+  };
+
+  void setDecoderVerbose(bool val) { mDecoderVerbose = val; };
+  void setDecoderBuffer(char* val) { mDecoderBuffer = val; };
```

Comment:
Since this is an input data buffer, should it be ``const char*``? I see that later to remove the constness of the input to pass it here, why?

Reply:
Hi @shahor02 ,
I am not sure I remember why it is like this.
Probably because when I started developing this code I did not use a `const char*` and this got propagated till today. Hence the need of the const cast below, because eventually I hit a place where I was delivered a `const char*`. I will take care of carefully checking the algorithm and bring `const char*` into place, as I agree this should be the way to do.